tests/Frontend/PathSpec.hs

```diff
@@ -1,23 +1,50 @@
+{-# LANGUAGE GADTs #-}
+
 {-# OPTIONS_GHC -Wall -Werror #-}
 
 module Frontend.PathSpec
 where
 
+import Data.String.Conversions (cs)
+import Data.Typeable (Typeable, typeOf)
+import Servant.API (FromHttpApiData(parseUrlPiece))
 import Test.Hspec (Spec, describe, it)
-import Test.QuickCheck (Gen, forAll, property)
+import Test.QuickCheck (Arbitrary, Gen, forAll, property)
 import qualified Data.Text as ST
 
 import Arbitrary
 import Data.UriPath
 import Frontend.Path
+import Types
 
 spec :: Spec
 spec = do
+
     describe "HasPath" $ do
         it "absoluteUriPath is not empty and well defined" . property . forAll mainGen $ \path ->
             ST.length (absoluteUriPath $ relPath path) >= 1
         it "relativeUriPath is not empty and well defined" . property . forAll mainGen $ \path ->
             ST.length (relativeUriPath $ relPath path) >= 0
+
+    describe "FromHttpApiData <-> UriPath" $ do
+        mapM_ uriPartAndHttpApiDataAreInverses
+            [ U (arb :: Gen PermissionContext)
+            , U (arb :: Gen IdeaSpace)
+            ]
   where
     mainGen :: Gen Main
     mainGen = arbitrary
+
+    arb :: Arbitrary a => Gen a
+    arb = arbitrary
```

Comment:
Total nitpick: it's defined already for the 3rd time. Perhaps it's worth exposing it in Arbitrary, to limit noise.